### PR TITLE
feat(core): add rotate and polarLerp methods to vector

### PR DESCRIPTION
**Pull request by ksassnowski**

### CODE REVIEW

File: packages/core/src/types/Vector.ts [hunk 1]
```diff
@@ -93,6 +94,91 @@ export class Vector2 implements Type {
...
+   * );
+   *
+   * // Rotating an object counterclockwise around the origin
+   * circle().position(
+   *   circle().position().rotate(180),
+   *   1,
+   *   easeInOutCubic,
+   *   Vector2.createPolarLerp(true),
+   * );
+   * ```
```
Comment (aarthificial) [inline packages/core/src/types/Vector.ts:137]: Multiple examples should be provided using multiple `@example` tags ([I'm not making this up](https://tsdoc.org/pages/tags/example/)):
```suggestion
   * @example
   * Animating an object in a circle around the origin
   * ```tsx
   * circle().position(
   *   circle().position().rotate(180),
   *   1,
   *   easeInOutCubic,
   *   Vector2.polarLerp
   * );
   * ```
   * @example
   * Rotating an object around the point [-200, 100]
   * ```ts
   * circle().position(
   *   circle().position().rotate(180, [-200, 100]),
   *   1,
   *   easeInOutCubic,
   *   Vector2.createPolarLerp(false, [-200, 100]),
   * );
   * ```
   * @example
   * Rotating an object counterclockwise around the origin
   * ```ts
   * circle().position(
   *   circle().position().rotate(180),
   *   1,
   *   easeInOutCubic,
   *   Vector2.createPolarLerp(true),
   * );
   * ```
```

**Review by aarthificial — CHANGES_REQUESTED** [review]

File: packages/core/src/types/Vector.ts [hunk 2]
```diff
@@ -93,6 +94,91 @@ export class Vector2 implements Type {
...
+      magnitude * Math.cos(angle) + origin.x,
+      magnitude * Math.sin(angle) + origin.y,
+    );
+  }
+
+  /**
+   * Helper function to create a {@link Vector2.polarLerp} interpolation
+   * function with additional parameters.
+   *
+   * @param counterclockwise - Whether the point should get rotated
```
[the remark this answers is not in the document]
Comment (ksassnowski) [reply packages/core/src/types/Vector.ts:174]: Thanks

File: packages/core/src/types/Vector.ts [hunk 3]
```diff
@@ -93,6 +94,95 @@ export class Vector2 implements Type {
...
+   * ```tsx
+   * circle().position(
+   *   circle().position().rotate(180),
+   *   1,
+   *   easeInOutCubic,
+   *   Vector2.polarLerp
+   * );
+   * ```
+   * @example
+   * Rotating an object around the point [-200, 100]
```
Comment (aarthificial) [inline packages/core/src/types/Vector.ts:123]:
```suggestion
   * Rotating an object around the point `[-200, 100]`
```

**Review by aarthificial — CHANGES_REQUESTED** [review]

**Review by aarthificial — APPROVED** [review]

**Pull request merged**